actors/evm/src/interpreter/instructions/call.rs

```diff
@@ -203,8 +205,7 @@ pub fn call<'r, BS: Blockstore, RT: Runtime<BS>>(
     let output_usize = if output_size.bits() < 32 {
         output_size.as_usize()
     } else {
-        // XXX that's probably a bug, should we barf instead?
-        1 << 31
+        return Err(StatusCode::InvalidMemoryAccess);
```

Comment:
This is nicer:

```
    // copy return data to output region if it is non-zero
    let output_usize = (output_size.bits() < 32)
        .then(|| output_size.as_usize())
        .ok_or(StatusCode::InvalidMemoryAccess)?;
```

BTW -- we don't support exact 32-bit lengths? Is this intentional?

Reply:
thats the entire wasm address space, we will panic anyway.

Reply:
From @vyzo:

> Our entire memory address space is 32 bit and a bunch of it is already taken, so I limit at 31 bit full (2G) to avoid headaches

This makes sense, breaks compatibility with Ethereum but in a reasonable way. Points to note here:

1. We don't have memory expansion gas yet.
2. We probably want a limit lower than 2GiB.

Reply:
@vyzo can you please add a TODO linking to https://github.com/filecoin-project/ref-fvm/issues/851?

Reply:
yessir.

Also note that `as_usize` panics if it doesnt fit.

Reply:
added TODO comment in https://github.com/filecoin-project/builtin-actors/pull/621/commits/001860c4a66ddc8524f364333f42bb33fc9c9dec